getferdi/ferdi
# Add notification & audio toggle action in tray context menu

Provide a new context menu item in the Tray bar that enables uses to mute / unmute the application . This provides the same functionality as enabling / disabling notifications and sound that is already available in the application.

### Description
Add new tray menu item to enable / disable Notifications and Audio
  - Use ipcMain to listen to iniital App Settings which determines the current isAppMuted state and display menu item text accordingly
  - Use ipcMain to listen to updates to App Settings -> isAppMuted from the App Menu / Side Bar and display menu item text accordingly
  - When the user clicks on the Enable / Disable Notifications & Audio menu item in the Tray, pass a message to the app via window.webContents.send with channel 'muteApp'
  - In stores/AppStore.js, use an ipcRenderer to toggleMuteApp()
  - To get the initial app state to the main process, pass a channel 'initialAppSettings' from SettingsStore which the tray class can listen to.

### Motivation and Context

The motivation behind this new feature is #520. It allows the user to mute/unmute the app from the system tray

The issue we are trying to address is #520 

### How Has This Been Tested?

Manual testing was done
1. When the app launches, made sure that the previous setting of isAppMuted is loaded in the tray bar
2. When clicking on the tray menu item, toggle the mute functionality. Ensure this changes the UI in the side bar as well as application main menu (services)
3. When muting / unmuting the app from the side bar or the Services menu, make sure the changes are reflected in the tray menu item.


1. Testing was done in Windows 10 on a development build



### Screenshots (if appropriate):
![image](https://user-images.githubusercontent.com/1255523/78597288-e01bb480-786a-11ea-8fd7-a95d35aea824.png)

![image](https://user-images.githubusercontent.com/1255523/78597323-f1fd5780-786a-11ea-9f24-644b9ca7e988.png)

### Types of changes


### Checklist:


- [x] My code follows the code style of this project (run `$ yarn lint`).
